src/components/biggive-search/biggive-search.tsx

```diff
@@ -9,17 +9,23 @@ import { faMagnifyingGlass, faX } from '@fortawesome/free-solid-svg-icons';
 export class BigGiveSearch {
   private searchText: string;
 
-  handleSearchTextChanged(event) {
+  private handleSearchTextChanged(event) {
     this.searchText = event.target.value;
   }
 
-  search() {
-    console.log(this.searchText);
+  /**
+   * This prop points to the memory address of the *real* search function
+   * on the parent of this component.
+   */ 
+  @Prop() doSearch: Function;
```

Comment:
There's an interesting answer about this [here](https://stackoverflow.com/a/64052525/2803757) @alihejazi 

It suggests that while this works in some contexts, we might hit problems passing a fn in others. This might be OK for components intended only for the Angular app but perhaps we should document that this is the case at component level when we make use of the approach.

I think it would also be good for another early component to use an `@Event`s approach so we can get a feel for the pro's and con's and decide whether to standardise on one approach or if it's OK to mix and match.

Reply:
@NoelLH Interesting you brough this up, because prior to pushing this change I had actually tried the `@Event` approach as you described, and it _did_ indeed work nicely but I thought you'd tell me it was over-engineering the logic, and that the approach I've pushed would be better.
I agree we should try the `@Event` approach for sure but let's do so with another component we'll be building, as this one is already coded up and I don't want to spend more time on it. This way, as you say, we can compare the two.

Reply:
I don't recall reviewing this specific bit @alihejazi but I only found out about the limitations in that link while reviewing this PR.

I don't have strong feelings about what we do first providing it works. However clicking the button seems to crash Storybook at the moment, which suggests we need at least some change before this version of the component works well enough to be part of our usage demo / style guide.

Reply:
@NoelLH implementation has now changed to emiting a custom event called `doSearch`.